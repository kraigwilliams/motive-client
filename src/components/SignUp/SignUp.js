import React, { Component } from 'react'
import {Required, FormLabel, FormInput, FormTitle } from '../Form/Form'
import AuthApiService from '../../services/auth-api-service'
import {FormButton, GoBack} from '../Button/Button'
import { FormWrapper, SignUpWrapper } from './SignUp.style'


export default class SignUp extends Component {
  static defaultProps = {
    history: {
      push: () => {},
    },
  }

  state = {
    error: null
  }

  handleRegistrationSuccess = () => {
    const { history } = this.props
    history.push('/login')
  }

  firstInput = React.createRef()

  handleSubmit = ev => {
    ev.preventDefault()
    const { firstname, lastname, username, password } = ev.target;
    AuthApiService.postUser({
      first_name: firstname.value, 
      last_name: lastname.value,
      username: username.value, 
      password: password.value,
    })
      .then(user => {
        firstname.value = ''
        lastname.value = ''
        username.value = ''
        password.value = ''
        this.handleRegistrationSuccess();
      })
      .catch(err => {
        this.setState({ error: err.error})
      })
  }

  componentDidMount() {
    this.firstInput.current.focus()
  }

  handleChange() {
    this.setState({
      error: null
    })
  }

  render() {
    const { error } = this.state
    return(
      <SignUpWrapper>
        
        <FormWrapper onSubmit={this.handleSubmit} onChange={this.handleChange.bind(this)}>
          <div>
            <GoBack 
              type='reset' 
              onClick={() => this.props.history.push('/')}
            />
            
          </div>
          <FormTitle>
            Create an account
          </FormTitle>
          {error && <p style={{textAlign: 'center', marginTop: '10px'}}>{error}</p>}
          <FormLabel htmlFor='signup-firstname-input'>
            First Name <Required />
          </FormLabel>
          <FormInput 
            ref={this.firstInput}
            placeholder='John'
            id='signup-firstname-input'
            name='firstname'
            aria-label="Enter your first name"
            aria-required="true"
            required
          />

          <FormLabel htmlFor='signup-lastname-input'>
            Last Name <Required /> 
          </FormLabel>
          <FormInput 
            placeholder='Doe'
            id='signup-lastname-input'
            name='lastname'
            aria-label="Enter your last name"
            aria-required="true"
            required
          />

          <FormLabel htmlFor='signup-username-input'>
            Username <Required /> 
          </FormLabel>
          <FormInput 
            placeholder='john_doe'
            id='signup-username-input'
            name='username'
            aria-label="Enter your username"
            aria-required="true"
            required
          />

          <FormLabel htmlFor='signup-password-input'>
            Password <Required />
          </FormLabel>
          <FormInput 
            type='password'
            placeholder='password'
            id='signup-password-input'
            name='password'
            aria-label="Enter your password"
            aria-required="true"
            required
          />

          <FormButton type='submit'>
            Sign Up
          </FormButton>
      </FormWrapper >
      
    </SignUpWrapper>
    )
  }
}