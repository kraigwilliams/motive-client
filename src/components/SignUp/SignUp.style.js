import styled from "styled-components";
import { colors } from "../constants";

export const FormWrapper = styled.form`
  display: flex;
  flex-direction: column;
  padding: 10px;
  background-color: ${colors.darkgrey};
  width: 100%;
  margin: auto;

  @media (min-width: 700px) {
    width: 70%;
    margin: auto;
    padding: 30px 50px;
    border-radius: 30px;
  }
`;

export const SignUpWrapper = styled.div`
  background-color: ${colors.darkgrey};
  margin: 0;
  padding: 40px;
  min-height: 100vh;

  @media (min-width: 700px) {
    background-color: ${colors.offwhite};
  }
`;
