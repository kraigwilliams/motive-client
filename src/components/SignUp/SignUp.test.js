import React from "react";
import ReactDOM from "react-dom";
import { mount, configure } from "enzyme";
import toJson from "enzyme-to-json";
import SignUp from "./SignUp";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });
describe(`<SignUp />`, () => {
  //Smoke Testing
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<SignUp />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  //Snapshot Testing
  it("renders SignUp by default", () => {
    const output = mount(<SignUp />);
    expect(toJson(output)).toMatchSnapshot();
  });
});
