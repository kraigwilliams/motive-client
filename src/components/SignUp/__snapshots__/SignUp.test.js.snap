// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUp /> renders SignUp by default 1`] = `
<SignUp
  history={
    Object {
      "push": [Function],
    }
  }
>
  <styled.div>
    <div
      className="sc-fzqNJr bphCiI"
    >
      <styled.form
        onChange={[Function]}
        onSubmit={[Function]}
      >
        <form
          className="sc-fzqBZW kKvITr"
          onChange={[Function]}
          onSubmit={[Function]}
        >
          <div
            style={
              Object {
                "height": "20px",
              }
            }
          >
            <Styled(Component)
              margin="0px"
              onClick={[Function]}
              type="reset"
            >
              <ForwardRef
                className="sc-fzoLsD esUdBY"
                margin="0px"
                onClick={[Function]}
                type="reset"
              >
                <button
                  className="Button sc-fzoLsD esUdBY"
                  margin="0px"
                  onClick={[Function]}
                  type="reset"
                >
                  <FontAwesomeIcon
                    border={false}
                    className=""
                    fixedWidth={false}
                    flip={null}
                    icon="arrow-left"
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    swapOpacity={false}
                    symbol={false}
                    title=""
                    transform={null}
                  >
                    <svg
                      aria-hidden="true"
                      className="svg-inline--fa fa-arrow-left fa-w-14 "
                      data-icon="arrow-left"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      style={Object {}}
                      viewBox="0 0 448 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
                        fill="currentColor"
                        style={Object {}}
                      />
                    </svg>
                  </FontAwesomeIcon>
                </button>
              </ForwardRef>
            </Styled(Component)>
          </div>
          <styled.h1>
            <h1
              className="sc-AxirZ gSjuDs"
            >
              Create an account
            </h1>
          </styled.h1>
          <Styled(Label)
            htmlFor="signup-firstname-input"
          >
            <Label
              className="sc-AxhUy iPWEFB"
              htmlFor="signup-firstname-input"
            >
              <label
                className="Label sc-AxhUy iPWEFB"
                htmlFor="signup-firstname-input"
              >
                First Name 
                <Required>
                  <span
                    className="Required"
                  >
                    *
                  </span>
                </Required>
              </label>
            </Label>
          </Styled(Label)>
          <Styled(Component)
            aria-label="Enter your first name"
            aria-required="true"
            id="signup-firstname-input"
            name="firstname"
            placeholder="Ralph"
            required={true}
          >
            <ForwardRef
              aria-label="Enter your first name"
              aria-required="true"
              className="sc-AxgMl gdCPlS"
              id="signup-firstname-input"
              name="firstname"
              placeholder="Ralph"
              required={true}
            >
              <input
                aria-label="Enter your first name"
                aria-required="true"
                className="Input sc-AxgMl gdCPlS"
                id="signup-firstname-input"
                name="firstname"
                placeholder="Ralph"
                required={true}
                type="text"
              />
            </ForwardRef>
          </Styled(Component)>
          <Styled(Label)
            htmlFor="signup-lastname-input"
          >
            <Label
              className="sc-AxhUy iPWEFB"
              htmlFor="signup-lastname-input"
            >
              <label
                className="Label sc-AxhUy iPWEFB"
                htmlFor="signup-lastname-input"
              >
                Last Name 
                <Required>
                  <span
                    className="Required"
                  >
                    *
                  </span>
                </Required>
              </label>
            </Label>
          </Styled(Label)>
          <Styled(Component)
            aria-label="Enter your last name"
            aria-required="true"
            id="signup-lastname-input"
            name="lastname"
            placeholder="Emerson"
            required={true}
          >
            <ForwardRef
              aria-label="Enter your last name"
              aria-required="true"
              className="sc-AxgMl gdCPlS"
              id="signup-lastname-input"
              name="lastname"
              placeholder="Emerson"
              required={true}
            >
              <input
                aria-label="Enter your last name"
                aria-required="true"
                className="Input sc-AxgMl gdCPlS"
                id="signup-lastname-input"
                name="lastname"
                placeholder="Emerson"
                required={true}
                type="text"
              />
            </ForwardRef>
          </Styled(Component)>
          <Styled(Label)
            htmlFor="signup-username-input"
          >
            <Label
              className="sc-AxhUy iPWEFB"
              htmlFor="signup-username-input"
            >
              <label
                className="Label sc-AxhUy iPWEFB"
                htmlFor="signup-username-input"
              >
                Username 
                <Required>
                  <span
                    className="Required"
                  >
                    *
                  </span>
                </Required>
              </label>
            </Label>
          </Styled(Label)>
          <Styled(Component)
            aria-label="Enter your username"
            aria-required="true"
            id="signup-username-input"
            name="username"
            placeholder="ralph_emerson"
            required={true}
          >
            <ForwardRef
              aria-label="Enter your username"
              aria-required="true"
              className="sc-AxgMl gdCPlS"
              id="signup-username-input"
              name="username"
              placeholder="ralph_emerson"
              required={true}
            >
              <input
                aria-label="Enter your username"
                aria-required="true"
                className="Input sc-AxgMl gdCPlS"
                id="signup-username-input"
                name="username"
                placeholder="ralph_emerson"
                required={true}
                type="text"
              />
            </ForwardRef>
          </Styled(Component)>
          <Styled(Label)
            htmlFor="signup-password-input"
          >
            <Label
              className="sc-AxhUy iPWEFB"
              htmlFor="signup-password-input"
            >
              <label
                className="Label sc-AxhUy iPWEFB"
                htmlFor="signup-password-input"
              >
                Password 
                <Required>
                  <span
                    className="Required"
                  >
                    *
                  </span>
                </Required>
              </label>
            </Label>
          </Styled(Label)>
          <Styled(Component)
            aria-label="Enter your password"
            aria-required="true"
            id="signup-password-input"
            name="password"
            placeholder="transcend1"
            required={true}
            type="password"
          >
            <ForwardRef
              aria-label="Enter your password"
              aria-required="true"
              className="sc-AxgMl gdCPlS"
              id="signup-password-input"
              name="password"
              placeholder="transcend1"
              required={true}
              type="password"
            >
              <input
                aria-label="Enter your password"
                aria-required="true"
                className="Input sc-AxgMl gdCPlS"
                id="signup-password-input"
                name="password"
                placeholder="transcend1"
                required={true}
                type="password"
              />
            </ForwardRef>
          </Styled(Component)>
          <Styled(Component)
            type="submit"
          >
            <ForwardRef
              className="sc-fzplWN dqDpy"
              type="submit"
            >
              <button
                className="Button sc-fzplWN dqDpy"
                type="submit"
              >
                Sign Up
              </button>
            </ForwardRef>
          </Styled(Component)>
        </form>
      </styled.form>
    </div>
  </styled.div>
</SignUp>
`;
